js/bars.js

```diff
@@ -1458,35 +1495,85 @@ historian.Bars.prototype.createSensorTable_ = function(values) {
+    var startTime = v[0];
     var endMs = parseInt(v[3],10);
+    var endTime = v[2];
     var durationOutput = historian.time.formatDuration(endMs - startMs);
 
-    // Highlight the row related to active connection.
-    if (v[v.length - 1] == "isActiveConn") {
-      highlightActiveConn.push(index);
-    }
     // Batching Period = -1 is a default value set for active connections 
     // without a history.
-    var batching = (v[9] == "-1.00") || (v[9] == "0.00")? "Not batching" : v[9];
+    var batching = v[9];
+    if (v[9] == "-1.00") {
+      batching = "Unavailable";
+    }
 
     var samplingRate = v[8]
-    if ((v[5].includes("ON_CHANGE")) && samplingRate == "-1.00") {
-      samplingRate = "on-change"
+    if (samplingRate == "-1.00") {
+      samplingRate = "Sampling Period\nis 0 second";
     }
 
-    return headRow.length > 6 ? 
-    [v[0], v[2], v[6], v[7], samplingRate, batching, v[10], durationOutput] :
-    [v[0], v[2], v[6], v[7], v[10], durationOutput];
+    var isActive = false;
+    if (v[v.length - 1] == "isActiveConn") {
+      highlightActiveConn.push(index);
+      isActive = true;
+      endTime = "Running";
+      durationOutput = "Active Connection";
+      // For active connection, we also show the actual sampling rate and 
+      // batching period information if available.
+      if (headRow.length == 7) {
+        headRow.push('Actual Sampling\nRate (Hz)');
+        headRow.push('Actual Batching\nPeriod (s)');
+      }
+      var sensorNum = v[4];
+      var sensorObj = getSensorByNumber(sensorNum);
+      var actualSamplingRate = sensorObj.ActualSamplingRateHz;
+      var actualBatchingPeriod = sensorObj.ActualBatchingPeriodS;
+      if (!actualBatchingPeriod) {
+        actualBatchingPeriod = "Not batching";
+      }
+      if (actualSamplingRate == "-1.00") {
+        actualSamplingRate = "(Not Applicalbe)\nSampling Period\nis 0 second";
+      }
+    }
+
+    // Do not show the start time and duration for subscription event with
+    // no activation statement. Also the requested sampling rate and batching
+    // period information will be missing.
+    if (v[v.length - 1] == "NoActivation") {
+      startTime = "Unknown";
+      durationOutput = "No activation\nfound in history";
+      samplingRate = "Unavailable";
+      batching = "Unavailable";
+    }
+
+    var singleRow;
+
+    if (headRow.length == 5) {
+      singleRow = [startTime, endTime, v[6], v[7], durationOutput];
+    } else if (isActive) {
+      singleRow = [startTime, endTime, v[6], v[7], samplingRate, batching, 
+        durationOutput,actualSamplingRate, actualBatchingPeriod];
+    } else {
+      singleRow = [startTime, endTime, v[6], v[7], samplingRate, batching, 
+        durationOutput];
+    }
+    // It is possible that multiple events with the same values are created, 
+    // then the entry.count field will record the number of events having this
+    // entry value.We want to show all events even though the values 
```

Comment:
Space after period.

Reply:
If possible about 1st person pronouns, I, We, etc

You can say "Show all events ..."